commons/src/main/java/io/hyscale/commons/utils/HyscaleFilesUtil.java

```diff
@@ -19,167 +19,210 @@
-     * @return
-     */
-
-    public String getFileName(String filePath) throws HyscaleException {
-        if (StringUtils.isBlank(filePath)) {
-            throw new HyscaleException(CommonErrorCode.EMPTY_FILE_PATH);
-        }
-
-        if (filePath.substring(filePath.length() - 1, filePath.length()).equals(FILE_SEPARATOR)) {
-            throw new HyscaleException(CommonErrorCode.FOUND_DIRECTORY_INSTEAD_OF_FILE, filePath);
-        }
-
-        int lastDirIndex = filePath.lastIndexOf(FILE_SEPARATOR);
-        if (lastDirIndex >= 0) {
-            return filePath.substring(lastDirIndex + 1, filePath.length());
-        } else {
-            return filePath;
-        }
-    }
-
-    /**
-     * Clears directory including internal dirs
-     *
-     * @param dir
-     * @throws HyscaleException
-     */
-    public void clearDirectory(String dir) throws HyscaleException {
-        File directory = new File(dir);
-        if (directory.isDirectory()) {
-            // true only if file path is directory and exists
-            try {
-                FileUtils.cleanDirectory(directory);
-            } catch (IOException e) {
-                HyscaleException ex = new HyscaleException(e, CommonErrorCode.FAILED_TO_CLEAN_DIRECTORY, dir);
-                throw ex;
-            }
-        }
-    }
-
-    public void deleteDirectory(String dir) throws HyscaleException {
-        File directory = new File(dir);
-        if (directory.isDirectory()) {
-            // true only if file path is directory and exists
-            try {
-                FileUtils.deleteDirectory(directory);
-            } catch (IOException e) {
-                HyscaleException ex = new HyscaleException(e, CommonErrorCode.FAILED_TO_DELETE_DIRECTORY, dir);
-                throw ex;
-            }
-        }
-    }
+	private static final Logger logger = LoggerFactory.getLogger(HyscaleFilesUtil.class);
+
+	private static final String NULL_STRING = "null";
+
+	/**
+	 * Create file in required directory
+	 *
+	 * @param filename
+	 * @param fileData
+	 * @throws HyscaleException
+	 */
+	public File createFile(String filename, String fileData) throws HyscaleException {
+		if (StringUtils.isBlank(filename)) {
+			throw new HyscaleException(CommonErrorCode.FAILED_TO_WRITE_FILE_DATA);
+		}
+		File file = createFile(filename);
+		if (StringUtils.isBlank(fileData)) {
+			logger.debug("Created empty file {}", filename);
+			return file;
+		}
+
+		try (FileWriter fileWriter = new FileWriter(file)) {
+			fileWriter.write(fileData);
+		} catch (IOException e) {
+			HyscaleException ex = new HyscaleException(e, CommonErrorCode.FAILED_TO_WRITE_FILE, filename);
+			throw ex;
+		}
+		return file;
+	}
+
+	/**
+	 * Create empty file with name
+	 * 
+	 * @param filename
+	 * @return
+	 * @throws HyscaleException if failed to create file
+	 */
+	public File createFile(String filename) throws HyscaleException {
+		File file = new File(filename);
+		return createFile(file);
+	}
```

Comment:
This can be merged into the same method below right ?

Reply:
It acts as a convenience method to classes calling the utility.